src/cls/_ZPM/PackageManager.cls

```diff
@@ -280,6 +293,138 @@ ClassMethod ShellInternal(pCommand As %String, Output pException As %Exception.A
 	}
 }
 
+/// d ##class(%ZPM.PackageManager).GetListModules("zpm",.list)
+ClassMethod GetListModules(StartsWith As %String = "", ByRef pList) [ Internal ]
+{
+	Set tQuery = "select Name,VersionString,Description,ExternalName from %ZPM_PackageManager_Developer.""MODULE"" where name %StartsWith ?"
+	Set tRes = ##class(%SQL.Statement).%ExecDirect(,tQuery,StartsWith)
+	
+	If (tRes.%SQLCODE < 0) {
+		Throw ##class(%Exception.SQL).CreateFromSQLCODE(tRes.%SQLCODE,tRes.%Message)
+	}
+	While tRes.%Next(.tSC) {
+		$$$ThrowOnError(tSC)
+		set pList(tRes.%Get("Name"))=$lb(tRes.%Get("VersionString"),tRes.%Get("Description"),tRes.%Get("ExternalName"))
+	}
+}
+
+/// Version client and registry
+ClassMethod Version(ByRef pCommandInfo) [ Internal ]
+{
+	new $Namespace
+	set $Namespace="%SYS"
+	
+	Set tQuery = "select Nsp from %SYS.Namespace_List(0,0) where Status = 1"
+	Set tRes = ##class(%SQL.Statement).%ExecDirect(,tQuery)
+	
+	If (tRes.%SQLCODE < 0) {
+		Throw ##class(%Exception.SQL).CreateFromSQLCODE(tRes.%SQLCODE,tRes.%Message)
+	}
+	While tRes.%Next(.tSC) {
+		$$$ThrowOnError(tSC)
+		set namespace(tRes.%Get("Nsp"))=""
+	}
+  
+  	Set tList = $$$HasModifier(pCommandInfo,"list")
+	if tList {
+		set ns=""
+		for { set ns=$order(namespace(ns)) q:ns=""
+			write !,"  "_ns_">",!
+			set $Namespace=ns
+			do ..Shell("list")
+		}
+	} else {
+		set $Namespace="%SYS"
+		do ..GetListModules("zpm",.list)
+		if $DATA(list("zpm")) {
+				write !," zpm "_$LG(list("zpm"),1)
+		}
+		set ns=""
+		for { set ns=$order(namespace(ns)) q:ns=""
+			kill list
+			set $namespace=ns
+			do ..GetListModules("zpm",.list)
+			if $DATA(list("zpm-registry")) {
+				write !,name_">zpm-registry "_$LG(list("zpm-registry"),1)
+				set Found=name
+			}
+		}
+		set $namespace="%SYS"
+		if '$d(Found) write !," zpm-registry not found" 
+		
+		// Get URL current registry
+		Set tRes = ##class(%SQL.Statement).%ExecDirect(,
+			"select Name "_
+			"from %ZPM_PackageManager_Client.ServerDefinition "_
+			"order by %ZPM_PackageManager_Client.ServerDefinition_SortOrder(ID) desc")
+		If (tRes.%SQLCODE < 0) {
+			Throw ##class(%Exception.SQL).CreateFromSQLCODE(tRes.%SQLCODE,tRes.%Message)
+		}
+		While tRes.%Next(.tSC) {
+			$$$ThrowOnError(tSC)
+			
+			#dim tDefn As %ZPM.PackageManager.Client.ServerDefinition
+			Set tRepository = ##class(%ZPM.PackageManager.Client.ServerDefinition).ServerDefinitionKeyOpen(tRes.%Get("Name"),,.tSC)
+			$$$ThrowOnError(tSC)
+			set repo($i(repo))=$lb(tRepository.Name,tRepository.Details)
+		}
+		$$$ThrowOnError(tSC)
+		set repo=""
+		for { set repo=$order(repo(repo)) q:repo=""
+			set url=$lg(repo(repo),2)
+			if $GET(url)'="" {
+				w !," ",url
+				// Get version registry
+				set tSC= ..GetJsonURL(url,.o)
+				$$$ThrowOnError(tSC)
+				w " "_o.version
+			}
+		}
+	}
+	q $$$OK
+}
+
+/// write $System.Status.GetErrorText(##class(%ZPM.PackageManager).GetJsonURL("https://pm.community.intersystems.com/",.o))
+ClassMethod GetJsonURL(url As %String, ByRef obj) As %Status
```

Comment:
I suggest moving GetJsonURL Method to class %ZPM.PackageManager.Client.REST.PackageManagerClient, where all methods to get data from the remote registry, is located.

Reply:
I moving GetJsonURL Method